### ISSUE TITLE
Houdini: Loading HDAs from tab menu

### ISSUE THREAD

**Issue by MustafaJafar** [ynput/ayon-houdini#9]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior:

By default Houdini doesn't install the published HDAs until a user load them from load tool.
When an HDA is installed, it becomes available to be loaded via TAB menu.
HDAs loaded from TAB menu are not tracked by the load tool.

**Problem 1: inventory tool tells the wrong count.** 

Also, removing the instances from the inventory tool, doesn't uninstall the HDA.
Which means I can still able to create as much as HDAs and inventory tool won't know.

 
But tbh, when updating the definition of an HDA. it's set in the Houdini session for this node type. 
which means Houdini will automatically update any instance of that node. 
which also means that count table should be grayed out because updating one instance will also update all the instances (Houdini default).

**Problem 2: updating one HDA instance updates all the HDAs because we are updating the HDA definition. So, we need that limitation to be visible in the inventory tool.**

### Expected Behavior:

Personally, I expect :
1. Created HDAs via TAB menu should also created AYON extra attributes
2. Inventory tool should disable actions per instance children.
![image](https://github.com/ynput/ayon-core/assets/20871534/149185fc-f9f4-4b01-9820-c508221dfd62)

### Version

1.0.0

### What platform you are running on?

Windows

### Steps To Reproduce:

In Houdini: 
**Problem 1: inventory tool tells the wrong count.** 
1. Publish an HDA product
2. Load that HDA 
3. Crete that HDA from TAB menu
4. open inventory tool, you'll find count is one.

**Problem 2: updating one HDA instance updates all the HDAs because we are updating the HDA definition. So, we need that limitation to be visible in the inventory tool.**
1. Publish multiple versions of an HDA product
2. Load that HDA multiple times
3. Set version for one instance
4. You'll find all the HDAs are updated because we are updating the node definition itself.

> [!NOTE]  
> You'll find all the HDAs are updated even though the inventory tool will detect a wrong version for them (It's the same as https://github.com/ynput/ayon-core/issues/470)

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the bug report.

[cuID:[AY-5262](https://app.clickup.com/t/86bypmgxc)]

**Comment by BigRoy** [comment]
> **Problem 1: inventory tool tells the wrong count.**
> 
> Also, removing the instances from the inventory tool, doesn't uninstall the HDA. Which means I can still able to create as much as HDAs and inventory tool won't know.

This _could_ be solved by ON publish and saving of the HDA secretly imprint it with the parms needed for the containerization (like give it the representation id, etc.) so that if you just create it - it has those attributes.

**Comment by MustafaJafar** [comment]
So, I found a solution for this one. 

We can _programmatically_ add a `onCreated` script. (we only to run this code once, maybe we can add it to the HDA Creator plugin.)
here's a snippet: 
```python
import hou

hda_node = hou.node('/obj/AVALON_CONTAINERS/SOPS/hou_hdas_hou_hdas_Test_1')
hda_def = hda_node.type().definition()


hda_def.addSection(
    "OnCreated",
"""
print("Hey, I exist. Say my name:",  kwargs['node'])
"""
    )
hda_def.setExtraFileOption("OnCreated/IsPython", True)
```

and in that `onCreated` script we add some logic to add the necessary AYON parameters, it will replace this [logic](https://github.com/ynput/ayon-houdini/blob/45b780634b5ca4ab5ac8cdb55280c2325117b8a1/client/ayon_houdini/plugins/load/load_hda.py#L49-L59) in the loader.
 
![Image](https://github.com/user-attachments/assets/b66e3eb3-64e5-4c01-9e56-abb543484c87)

I'd recommend to use generic code in the `onCreated` script so that artists are free to use the HDA outside of AYON.

**Comment by BigRoy** [comment]
> I'd recommend to use generic code in the `onCreated` script so that artists are free to use the HDA outside of AYON.

This would also be possible if we embed the attributes on publish into the HDA itself, which is what I'd prefer to do if we can.

**Comment by MustafaJafar** [comment]
> you want active, per node.

I don't think it works per node. once changing the asset definition, in one node, it affects all of existing node as shown in #304

**Comment by BigRoy** [comment]
> I don't think it works per node. once changing the asset definition, in one node, it affects all of existing node as shown in [#304](https://github.com/ynput/ayon-houdini/issues/304)

Make HDAs regularly - in a single HDA, and upversion them.

1. Create HDA

<img width="714" height="565" alt="Image" src="https://github.com/user-attachments/assets/32c24018-7fb3-4cbe-895f-7e7d395915b4" />

2. Increase minor version

<img width="388" height="412" alt="Image" src="https://github.com/user-attachments/assets/b6a62f0b-f126-4345-9e2f-2c58e64a6f3e" />

<img width="542" height="575" alt="Image" src="https://github.com/user-attachments/assets/51750088-d182-4442-981f-fd68624a448c" />
Save it.

3. Add a parm via Type Properties

<img width="270" height="500" alt="Image" src="https://github.com/user-attachments/assets/37d6b134-c1f4-4c5d-b6b7-d8dbd541fd1a" />

---

So now you have a single 'node type' with multiple versions.
Now let's enable the "Version" bar... so it's easy to switch, go to asset manager:

<img width="331" height="470" alt="Image" src="https://github.com/user-attachments/assets/08782d03-1056-4482-91e3-feb68e88fb9d" />

To the configure tab and enable "Asset Definition Toolbar > Show Menu Always"

<img width="747" height="523" alt="Image" src="https://github.com/user-attachments/assets/c4f6ebea-0e5e-48cb-b4f8-48900315e082" />

Note how now you can switch that one HDA between the versions, attributes will appear/disappear on the switch.
Furthermore, you can have two nodes in your scene with different versions.

Voila.
